### DISCUSSION TITLE
Bug of user-not-logged-in in the recovery flow and solution with `link.config.base_url`

### DISCUSSION THREAD

**Discussion by hugotiburtino** [ory/kratos#2667] [Show and tell]
Since Kratos relies on cookies and cookies can only be set to its own domains, it would be quite a problem if your kratos instance runs in another domain as your application, often the case v.g. with next.js apps, which many like to deploy in vercel, or if you are testing it on localhost .

Fortunatelly, Ory has already the repo [ory/integrations](https://github.com/ory/integrations) which comes with a handy dandy `createApiHandler` function. Instead of pointing to your kratos instance, you can point to an endpoint containing this function (like in [example from Ory](https://github.com/ory/kratos-selfservice-ui-react-nextjs/blob/master/pages/api/.ory/%5B...paths%5D.ts), which gets the cookies from kratos and sets them for your own domain. It is thought to work in a next.js app, but as long as you may have dynamic paths (as in next.js it is for example  `path/[...slug].js`), I dare to say it may work in other apps, too. 

Now, it took me some time to understand that I should use it also in the recovery flow. After clicking in the recovery link, the user was redirected to the application, but they weren't logged in, not being able to change the password. I suppose that is the reason why Ory Team already thought about that and provided the configuration of `link.config.base_url`.

TL;DR
If the user cannot change the password in your next.js app, although all others flows work, change your configuration to:

```yaml
selfservice:
  methods:
    link:
      enabled: true
      config:
        base_url: https://yourapp.org/api/.ory
```

**Comment by aeneasr** [comment]
Thank you for the feedback! We are already working on a PR with a new recovery strategy which will working using a code sent to the email, instead of a magic link.

This will resolve the problems you describe :)
